arduino_watersystem_demo/esp32client/esp32client.ino

```diff
@@ -0,0 +1,194 @@
+#include <Arduino.h>
+#include <Hash.h>
+
+#include <WiFi.h>
+#include <WebSocketsClient.h>
+#include "SoftwareSerial.h"
+#include <ArduinoJson.h>
+
+const char* wlan_ssid             = "xiomi";
+const char* wlan_password         = "porc2232";
+const char* ws_host               = "127.0.0.1";
+const int   ws_port               = 8080;
+
+String roomName = "lala";
+String username = "esp32";
+
+SoftwareSerial s1(34, 35); // RX, TX
+
+String softwareSerialData = "";
+char character;
+
+// base URL for SockJS (websocket) connection
+// The complete URL will look something like this(cf. http://sockjs.github.io/sockjs-protocol/sockjs-protocol-0.3.3.html#section-36):
+// ws://<ws_host>:<ws_port>/<ws_baseurl>/<3digits>/<randomstring>/websocket
+// For the default config of Spring's SockJS/STOMP support, the default base URL is "/socketentry/".
+const char* ws_baseurl            = "/chat/"; // don't forget leading and trailing "/" !!!
+
+
+WebSocketsClient webSocket;
+
+
+void setup() {
+
+  Serial.begin(115200);
+  s1.begin(9600);
+  connectToWifi();
```

Comment:
You can implement a retry mechanism for connecting to Wifi, or use a loop. Maybe the connection will fail on the first try. You can also increase the delay time.